pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/OffloadPolicies.java

```diff
@@ -96,6 +95,56 @@ public String getValue() {
         }
     }
 
+    @InterfaceAudience.Public
+    @InterfaceStability.Stable
+    public enum OffloadMethod {
+        /**
+         * Ledger based offload, one offload segment corresponding to a ledger
+         */
+        LEDGER_BASED("ledger-based"),
+        /**
+         * Streaming offload, offload segments are divided by time or size
+         */
+        STREAMING_BASED("streaming-based"),
+        /**
+         * Disable offload
+         */
+        NONE("none");
```

Comment:
This can be used to reduce boilerplate code in usage, for example this method return NONE when offload policies is null.
```java
    public OffloadMethod getOffloadMethod() {
        if (config.getLedgerOffloader() == null) {
            return OffloadMethod.NONE;
        }
        if (config.getLedgerOffloader().getOffloadPolicies() == null) {
            return OffloadMethod.NONE;
        }
        return config.getLedgerOffloader().getOffloadPolicies().getOffloadMethod();
    }
```
so the caller can simply do a switch rather than have a complex if check

Reply:
The NONE for the enum does not meaningfully, and from the description, NONE represents offload disabled, this also does no make sense. If the default Offload method is LedgerBased, you return the LedgerBased as default.